### PR TITLE
Implementing fmod function.

### PR DESCRIPTION
**Pull request by tek1031**
This PR implements fmod with covering test for input of dtype float16, float32 and float64.
This fmod function requred cupy fix function.
So, I also implement cupy fix function.

### CODE REVIEW

**Comment by mattya** [comment]
Sorry for the late response.
I found the numerical gradient test is failed.
This is because the `fmod` function has discontinuities at fmod(x, d) equals to 0.
Also, the result will be unstable when divisor is too small.
Could you avoid these discontinuity points in the test code like [test_relu.py](https://github.com/pfnet/chainer/blob/master/tests/chainer_tests/functions_tests/activation_tests/test_relu.py#L25)?

And also please make small fix on `chainer/functions/__init__.py` and `docs/source/reference/functions.rst`.  They have some conflicts.

**Comment by tek1031** [comment]
Thank you for letting me know the conflict and test fail.
I fixed these issues.

File: chainer/functions/math/fmod.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+
+def fmod(x, divisor):
+    """Elementwise mod function.
+
+    .. math::
+       y_i = x_i mod div.
+
+    Args:
+        x (~chainer.Variable): Input variable.
```
Comment (mattya) [inline chainer/functions/math/fmod.py]: Please add a description  about `div`.

**Comment by mattya** [comment]
LGTM excepting above one.

**Comment by tek1031** [comment]
Sorry for the late response.
I changed comment.

**Comment by mattya** [comment]
Thank you for the update.
I'll merge it after the CI passes.

File: cupy/__init__.py [hunk 2]
```diff
@@ -316,6 +316,7 @@
 
 from cupy.math.rounding import ceil  # NOQA
 from cupy.math.rounding import floor  # NOQA
+from cupy.math.rounding import fmod  # NOQA
```
Comment (mattya) [inline cupy/__init__.py]: I think this line should be 
```python
from cupy.math.rounding import fix
```

File: chainer/functions/math/fmod.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    @property
+    def label(self):
+        return 'fmod'
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+        type_check.expect(
+            in_types.size() == 2,
+            in_types[0].dtype.kind == 'f',
```
Comment (unnonouno) [inline chainer/functions/math/fmod.py]: Do you need to check if `in_types[0].dtype == in_types[1].dtype`?

File: chainer/functions/math/fmod.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        gw, = grad_outputs
+        x, divisor = inputs
+        return gw, utils.force_array(xp.fix(x / divisor) * -1 * gw, x.dtype)
+
+
+def fmod(x, divisor):
+    """Elementwise mod function.
+
+    .. math::
+       y_i = x_i mod div.
```
Comment (unnonouno) [inline chainer/functions/math/fmod.py]: Is this LaTeX code correct?
`y_i = x_i \bmod d` ?

File: chainer/functions/math/fmod.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        gw, = grad_outputs
+        x, divisor = inputs
+        return gw, utils.force_array(xp.fix(x / divisor) * -1 * gw, x.dtype)
+
+
+def fmod(x, divisor):
+    """Elementwise mod function.
+
+    .. math::
+       y_i = x_i \bmod divisor.
```
Comment (unnonouno) [inline chainer/functions/math/fmod.py]: In LaTeX, `divisor` represents multiplication of `d`, `i`, `v`,`i`, `s`, `o` and `r`. Please use a single character `d` and append its description like ", where :math:`d` is a given divisor." or use `\\mathrm{divisor}`.

Comment (unnonouno) [inline chainer/functions/math/fmod.py]: And please use `\\bmod` to avoid escape in python world.

File: chainer/functions/math/fmod.py [hunk 6]
```diff
@@ -36,11 +36,11 @@ def fmod(x, divisor):
     """Elementwise mod function.
 
     .. math::
-       y_i = x_i \bmod divisor.
+       y_i = x_i \\bmod divisor.
```
Comment (unnonouno) [inline chainer/functions/math/fmod.py]: No, please fix here. In ``.. math::``

File: chainer/functions/math/fmod.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+def fmod(x, divisor):
+    """Elementwise mod function.
+
+    .. math::
+       y_i = x_i \\bmod \\mathrm{divisor}.
+
+    Args:
+        x (~chainer.Variable): Input variable.
+        \\mathrm{divisor} (~chainer.Variable): Input divisor.
```
Comment (unnonouno) [inline chainer/functions/math/fmod.py]: This is not LaTeX... 
```
divisor (~chainer.Variable): ...
```
Please make the document using Sphinx and check the result.

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**